### PR TITLE
Mark `indices.exists_type`/`TypeExists` APIs as deprecated.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**

### CODE REVIEW

File: src/OpenSearch.Client/Descriptors.Indices.cs [hunk 1]
```diff
@@ -482,6 +482,7 @@ protected IndexTemplateExistsDescriptor(): base()
 	}
 
 	///<summary>Descriptor for TypeExists <para>https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/exists/</para></summary>
+	///<remarks>Deprecated as of OpenSearch 2.0</remarks>
 	public partial class TypeExistsDescriptor : RequestDescriptorBase<TypeExistsDescriptor, TypeExistsRequestParameters, ITypeExistsRequest>, ITypeExistsRequest
```
Comment (acarbonetto) [inline src/OpenSearch.Client/Descriptors.Indices.cs:486]: Should we mark the class as deprecated?
https://stackoverflow.com/questions/314505/how-to-mark-a-class-as-deprecated

Comment (Yury-Fridlyand) [reply src/OpenSearch.Client/Descriptors.Indices.cs:486]: Unfortunately, no. There are few reasons, one of them: it is not deprecated for OpenSearch 1.0.
https://github.com/Bit-Quill/opensearch-net/pull/31#discussion_r913360046

**Review by MaxKsyunz — COMMENTED** [review]
@Yury-Fridlyand  can you double check that `/{index}/mapping/{type}` is behaving similarly between OpenSearch 1.x and OpenSearch 2.0?

As far as I can see, that endpoint is not available in OpenSearch 2.0.

**Comment by Yury-Fridlyand** [comment]
> @Yury-Fridlyand can you double check that `/{index}/mapping/{type}` is behaving similarly between OpenSearch 1.x and OpenSearch 2.0?
> 
> As far as I can see, that endpoint is not available in OpenSearch 2.0.

`/{index}/mapping/{type}` was marked as deprecated in 1.x and removed in 2.0.
The corresponding APIs were removed from .Net client in stage of adoption of Elastic .Net client. I just checked that nothing left. Even OpenSearch 1.0 client can't use these APIs.

**Review by acarbonetto — APPROVED** [review]

**Pull request merged**